src/main/java/httpserver/HttpRequest.java

```diff
@@ -0,0 +1,16 @@
+package httpserver;
+
+
+import java.util.*;
+
+public class HttpRequest implements HttpIncomingMessage{
+    private HashMap<String, String> attributes;
+
+    public HttpRequest(HashMap clientRequestAttributes){
```

Comment:
A `HashMap` is getting injected but we only use the `route` key. Is using a hash this a little premature?

Reply:
In the context of HTTP Requests, what is a *route*? Does the HTTP Spec call this a route?